WordPress/gutenberg
# Remove unnecessary `view.js` file from navigation-submenu block

## What?
If I am not mistaken, the current [`view.js` file in the navigation-submenu block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-submenu/view.js) is not being used. It isn't defined in the [block.json](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-submenu/block.json) (there is no viewScript field) and, if I add a submenu in the navigation block, it doesn't seem to load that view.js file in the browser.

Apart from that, it seems the [navigation block view.js file](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/view.js) is doing mostly the same and seems a bit more up-to-date. This one is loaded in the browser.

## Why?
If it is not being used, it is better to remove unnecessary code.

## How?
I just removed the mentioned file.

## Testing Instructions

1. Add a submenu item in the navigation block.
2. Check that everything works as expected. From what I understood, that file should keep the submenu opened when clicking the arrow, which is already handled by the parent `navigation` block.
